src/edu/washington/escience/myria/operator/Split.java

```diff
@@ -0,0 +1,146 @@
+package edu.washington.escience.myria.operator;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+
+import edu.washington.escience.myria.DbException;
+import edu.washington.escience.myria.Schema;
+import edu.washington.escience.myria.Type;
+import edu.washington.escience.myria.column.Column;
+import edu.washington.escience.myria.column.builder.ColumnBuilder;
+import edu.washington.escience.myria.column.builder.ColumnFactory;
+import edu.washington.escience.myria.storage.TupleBatch;
+import edu.washington.escience.myria.storage.TupleUtils;
+
+/**
+ * Generic apply operator.
+ */
+public class Split extends UnaryOperator {
+  /***/
+  private static final long serialVersionUID = 1L;
+
+  /**
+   * Name of column to split on regex. Must have String type.
+   */
+  private final String columnName;
+
+  /**
+   * Compiled regex to split input tuples.
+   */
+  private final Pattern pattern;
+
+  /**
+   * Schema of output tuples.
+   */
+  private Schema outputSchema;
+
+  /**
+   * Index of column to split in input schema (and anonymous result column in output schema).
+   */
+  private int splitColIdx;
+
+  /**
+   * Array of column indices in output schema corresponding to column indices of input schema. The column to be split
+   * has its index mapped to the sentinel value -1.
+   */
+  private int[] inputColIndexToOutputColIndex;
```

Comment:
I think you can simplify this logic. If we define the output of `split` to be like this:

Given input `R(a, b, c)`, `split(R, b)` should return a relation `(a, c, s1), (a,c,s2), ...`

Then we can just make this an array of all the indices that are not split. In this case, it would be `[0,2]` to represent `a` and `c`. Then the code below can ignore any sentinel checks and just prepend the columns listed herein.